### PR TITLE
Copy `fixed-data-table` to `fixed-data-table-2`

### PR DESCRIPTION
**Pull request by stevegreatrex**
Suggestion from maintainers of `fixed-data-table-2` (see https://github.com/schrodinger/fixed-data-table-2/issues/216) was to duplicate the type defs for the old project as a stop-gap until proper typings are created

I've copied the `fixed-data-table` folder to `fixed-data-table-2` and updated all references.

# Template

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
> n/a, as it is an exact copy of an existing project (?)
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by stevegreatrex** [comment]
I can't tick the box for `is correct naming convention` but that _is_ the correct name of the package!

File: types/fixed-data-table-2/tslint.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
```
Comment (plantain-00) [inline types/fixed-data-table-2/tslint.json]: For new packages, this file should only extends `"dtslint/dt.json"`, and no other rules here.

File: types/fixed-data-table-2/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+	"files": ["index.d.ts", "fixed-data-table-2-tests.tsx"],
+	"compilerOptions": {
+		"module": "commonjs",
+		"lib": ["es6", "dom"],
+		"noImplicitAny": true,
+		"noImplicitThis": true,
+		"strictNullChecks": false,
```
Comment (plantain-00) [inline types/fixed-data-table-2/tsconfig.json]: For new packages, `strictNullChecks` should be enabled here.

File: types/fixed-data-table-2/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,522 @@
+// Type definitions for fixed-data-table-2 copied from definitions for fixed-data-table 0.6.0
```
Comment (plantain-00) [inline types/fixed-data-table-2/index.d.ts]: You can just use: `// Type definitions for fixed-data-table-2`

**Review by plantain-00 — CHANGES_REQUESTED** [review]

**Comment by stevegreatrex** [comment]
Thanks for the notes!  I've updated the PR as requested.

After removing the TSLint rule definitions I noticed a few other issues with `index.d.ts` so I've updated those as well, with one exception: https://github.com/Microsoft/dtslint/blob/master/docs/export-just-namespace.md

This is flagged by the linter but I'm concerned that removing the reference to the namespace will affect backwards-compatibility with the `fixed-data-table` library from which this was copied.

The intention with this PR is to allow consumers of `fixed-data-table-2` (which was originally a copy of `fixed-data-table`) to use the new package as a drop-in replacement.  If we modify the exported namespace I _believe_ this could negatively affect consumer projects.

Thoughts?

**Comment by stevegreatrex** [comment]
Thanks for the notes! I've updated the PR as requested.

After removing the TSLint rule definitions I noticed a few other issues with index.d.ts so I've updated those as well

**Review by plantain-00 — APPROVED** [review]

File: types/fixed-data-table-2/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,515 @@
+// Type definitions for fixed-data-table-2 0.8
+// Project: https://github.com/facebook/fixed-data-table-2
```
Comment (armanio123) [inline types/fixed-data-table-2/index.d.ts]: Project should make reference to https://github.com/schrodinger/fixed-data-table-2

**Review by armanio123 — CHANGES_REQUESTED** [review]

**Comment by armanio123** [comment]
PR #21722 appears to be for the same library. Could you please indicate if there's a difference on the implementation provided by @ilivit?

**Comment by ilivit** [comment]
To be noted, the typings in PR #21722 are based on propTypes from [schrodinger/fixed-data-table-2](https://github.com/schrodinger/fixed-data-table-2). These APIs are slightly different (i.e. some new props have been added).

**Comment by stevegreatrex** [comment]
Ok, it looks like #21722 is a better set of typings so I'll close this one.

**Pull request closed without merging**